docroot/profiles/custom/sitenow/sitenow.install

```diff
@@ -1950,3 +1951,116 @@ function sitenow_update_9054() {
+    $bundles = \Drupal::service('entity_type.bundle.info')
+      ->getBundleInfo($entity_type);
+
+    // @todo Handle case where no bundles exist?
+    foreach ($bundles as $bundle => $bundle_label) {
+      /** @var \Drupal\Core\Entity\EntityFieldManagerInterface $entityFieldManager */
+      $entityFieldManager = \Drupal::service('entity_field.manager');
+      $fields = $entityFieldManager->getFieldDefinitions($entity_type, $bundle);
+      $field_check = [];
+      /**
+       * @var string $field_name
+       * @var \Drupal\field\Entity\FieldConfig $definition
+       */
+      foreach ($fields as $field_name => $definition) {
+        if (in_array($definition->getType(), [
+          'text_long',
+          'text_with_summary',
+        ])) {
+          $field_check[$field_name] = $definition;
+        }
+      }
+
+      // Only need to run the following if we found fields to check against.
+      if (!empty($field_check)) {
+
+        // Retrieve all entities of this bundle type;.
+        // Media is listed as 'bundle,' but other entity types
+        // are listed as 'type'.
+        $type = $entity_type == 'media' ? 'bundle' : 'type';
+        $entities = \Drupal::entityTypeManager()->getStorage($entity_type)
+          ->loadByProperties([$type => $bundle]);
+
+        // Loop through the loaded entities.
+        foreach ($entities as $entity) {
+          $needs_save = FALSE;
+
+          // Loop through the fields we defined before.
+          foreach ($field_check as $name => $definition) {
+            // If the field has a value (or values).
+            if (!$entity->get($name)->isEmpty()) {
+              $updated_values = [];
+              $field_values = $entity->{$name}->getValue();
+
+              foreach ($field_values as $field_value) {
+
+                // @todo Any special handling for text_with_summary fields?
+                $document = Html::load($field_value['value']);
+
+                // Get all <drupal-media> instances.
+                $dm_instances = $document->getElementsByTagName('drupal-media');
+
+                // Built with this as a template.
+                // https://github.com/uiowa/uiowa/pull/3048/commits/5b95c9912aaa2b235cf06787a8c673a0f2cb17d7
+                $i = $dm_instances->length - 1;
+
+                while ($i >= 0) {
+                  // The current inline image element.
+                  $old_dm_instance = $dm_instances->item($i);
+                  $new_dm_instance = $old_dm_instance;
+
+                  $view_mode = $old_dm_instance->getAttribute('data-view-mode');
+                  $is_relevant_view_mode = in_array($view_mode, $media_view_mode_map_keys);
+                  if ($is_relevant_view_mode) {
+                    $new_dm_instance->setAttribute('data-view-mode', $media_view_mode_map[$view_mode]);
+
+                    $old_dm_instance->parentNode->replaceChild($new_dm_instance, $old_dm_instance);
+                  }
```

Comment:
Should the `$needs_save = TRUE;` be moved up here? I think this is the point where we know a change is needed, and if we don't hit this, it's possible we had an entity with the right fields, values in those fields, but no embeds to be updated.